### PR TITLE
tests-common: Handle PortInUseException for SSHConsoleConn

### PR DESCRIPTION
**Pull request by matthew-soulsby**
### Description of PR

Addresses https://github.com/sonic-net/sonic-mgmt/issues/15092

Summary:
This PR adds handling for the PortInUseException, which is thrown when an SSHConsoleConn is successfully established (most prominently in dut_console tests) but the console port is occupied, resulting in the connection being terminated by the host.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
This PR is intended to add more resilience to the dut_console tests, as if a testbed with a blocked port was selected for a nightly test, it would cause the dut_console tests to fail during setup. This provides a method for auto-recovery for these cases.

#### How did you do it?
By completing the following:
* Add connection types for each of the configuration menu types (Digi, Cisco, and Sonic) to allow for different command sequences to be accounted for
* Add function to clear a DUT's console port
* Add retry logic to add resilience to main DUT connection set up 

#### How did you verify/test it?
This process was conducted on 5 testbeds, with at least one of each config menu type.

##### Steps conducted during testing

1. SSH into testbed using the console IP and console port - simulating a blocked port
2. In a separate terminal, run any dut_console test individually
3. During test setup, the connection from step 1 was terminated successfully
4. The test then runs successfully

##### Example test output
* Successful port reset (Digi config):

Screenshot 2024-10-25 165601
* Successful port reset (Sonic config):

Screenshot 2024-10-25 183646
* Sample failure 
  * Simulating config menu type is not defined (causing the port clear function to exit early), and a more descriptive error message is thrown regarding why the connection to the DUT is failing:

Screenshot 2024-10-25 153008

#### Any platform specific information?
N/A - this is a generalised solution which accounts for as many types of configuration menus as possible.

#### Supported testbed topology if it's a new test case?
N/A

### Documentation
N/A

### CODE REVIEW

File: ansible/library/conn_graph_facts.py [hunk 1]
```diff
@@ -349,6 +349,7 @@ def csv_to_graph_facts(self):
                     "peerport": entry["StartPort"],
                     "proxy": entry["Proxy"],
                     "type": entry["Console_type"],
+                    "menu_type": entry["Console_menu_type"],
```
Comment (ZhaohuiS) [inline ansible/library/conn_graph_facts.py:352]: @matthew-soulsby please also give an update for the example file `ansible/files/sonic_lab_console_links.csv`

Comment (matthew-soulsby) [reply ansible/library/conn_graph_facts.py:352]: Good point, added the column with empty values (to leave behaviour as it was prior to the changes), how does that look?

**Review by ZhaohuiS — APPROVED** [review]
LGTM

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Comment by vperumal** [comment]
@abdosi @yejianquan - Can we get this merged to 202405 - this is causing breakage because it is now looking for both console_type in some cases and console_menu_type in some cases. For now I have defined both

**Comment by matthew-soulsby** [comment]
Hi @vperumal, this was merged into 202405 in this PR (due to issues with the auto cherry-pick PR): https://github.com/sonic-net/sonic-mgmt/pull/15293

Did you have any logs regarding the issue you're facing? Happy to look into it if so

**Pull request merged**